### PR TITLE
Unit Tests for ShoppingListItemSpecification

### LINKED ISSUE #12: Create unit test to ShoppingListItemSpecification.java
### Create Unit Tests for `ShoppingListItemSpecification` (`GreenCityMVP`)

**File:**  
`service/src/main/java/greencity/filters/ShoppingListItemSpecification.java`

###  Description

Create a suite of unit tests for the `ShoppingListItemSpecification` class.  
It builds dynamic queries for `ShoppingListItem` entities based on a list of `SearchCriteria`, such as:

- Filter by `id` (numeric)
- Filter by `content` (via join with `ShoppingListItemTranslation`)
- Support for compound predicates

### Input Data

- Filter types used: `"id"`, `"content"`
- Criteria API classes involved:
  - `CriteriaBuilder`
  - `CriteriaQuery`
  - `Root<ShoppingListItem>`
- Uses Hibernate Metamodel:
  - `ShoppingListItem_`
  - `ShoppingListItemTranslation_`

### Test Scenarios

- [x] `toPredicate(...)` returns valid predicate for `"id"` search
- [x] `toPredicate(...)` returns valid predicate for `"content"` search using join with `ShoppingListItemTranslation`
- [x] Predicate skips filtering when `"content"` is empty or blank
- [x] Compound predicate (both `"id"` and `"content"` in list)
- [ ] Edge case: `searchCriteriaList` contains unknown `type` (should be ignored or tested)

###  Acceptance Criteria

- [x] All tests pass successfully (`JUnit 5`)
- [x] `CriteriaBuilder`, `CriteriaQuery`, `Root`, etc. are mocked with `Mockito`
- [x] All `if` conditions are covered
- [x] Private helper method `getTranslationPredicate(...)` is tested indirectly
- [x] All paths (including empty/blank `content`) are covered
- [x] **100% test coverage** with both line and branch coverage

**Tip**:  
Simulate `SearchCriteria` with both `id` and `content`, and verify that `criteriaBuilder` methods are invoked as expected.

Mock static metamodel classes like `ShoppingListItem_` only if required (or consider simplifying with string-based field names for test purposes).

### PR DESCRIPTION
**Pull request by Shcherba04**
## Summary by CodeRabbit

* **Tests**
  * Added comprehensive tests for filtering and searching shopping list items, ensuring correct behavior when applying various search criteria.

### CODE REVIEW

File: service/src/test/java/greencity/filters/ShoppingListItemSpecificationTest.java [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+public class ShoppingListItemSpecificationTest {
+    @Mock
+    private Root<ShoppingListItem> ShoppingListItemRoot;
+
+    @Mock
+    private CriteriaBuilder builder;
+
+    @Mock
+    private CriteriaQuery MockCriteriaQuery;
+
```
Comment (AndriiSynytsia) [inline service/src/test/java/greencity/filters/ShoppingListItemSpecificationTest.java]: Public access classifier is not necessary in tests. Double check the naming of your variables and instances They should be in **camelCase**. Use formatting to clean up spacing in your code.

**Review by AndriiSynytsia — CHANGES_REQUESTED** [review]
Hey. Hope you are doing well. There is some changes that would be nice to fix. Also consider about applying @BeforeAll preset for your tests e.g. `when(builder.conjunction()).thenReturn(conjunctionPredicate);`

**Review by Shcherba04 — COMMENTED** [review]
camelCase and spacing in tests

**Comment by AndriiSynytsia** [comment]
Check the review from Rabbit, it highlighted for you couple improvements. Change it when you will have time and reach me out :)

**Review by khomenkooleksandr — CHANGES_REQUESTED** [review]
Check the review from Rabbit, it highlighted for you couple improvements.
